gensim/models/tfidfmodel.py

```diff
@@ -6,51 +6,103 @@
 
 
 import logging
-import math
+from functools import partial
 
 from gensim import interfaces, matutils, utils
 from six import iteritems
 
+import numpy as np
 
 logger = logging.getLogger(__name__)
 
 
+def resolve_weights(smartirs):
```

Comment:
docstrings needed too (for all stuff here)

Reply:
I think that `Checks for validity of smartirs parameter.` is enough. Do you have anything else in mind as well?

Reply:
@markroxor need to add "Parameters" (type, description), "Raises" (type, reason), "Returns" (type, description)